scipy/signal/wavelets.py

```diff
@@ -308,14 +309,147 @@ def ricker(points, a):
     return total
 
 
-def cwt(data, wavelet, widths):
+def morlet2(M, s, w=5):
```

Comment:
It's a bit odd that the signature of ``morlet`` has ``s, w`` reversed compared to this. Why not make that consistent?

Reply:
In a bit more detail, the `w` parameter in `morlet(M, w=5.0, s=1.0, complete=True)` is not the width, but `cwt` has an absolute requirement for the second parameter in wavelet functions to correspond to the width.

It might be possible to use the same letters for the parameters in `morlet2()`, but assign them different meaning. Is that really clearer though?

Playing around with this a bit locally gets messy quick. The canonical documented example of an appropriately propotyped wavelet function is `ricker(points, a)`, with two positional arguments & no keyword arguments, and a different letter for width.

Reply:
I think there's some confusion about the nomenclature here. It's been quite some time since I've dealt with this properly, and my understanding of the topic might be flawed in one way or another. I'll reference an excellent source on this topic [1], which I hope can make things clearer.
  
The wavelet function implemented here is basically a frequency modulated Gaussian window, normalized to unit norm ([1], Section 4.3), which is normally referred to as the Gabor wavelet ([1], Eq. (4.62)), but  is also often called the Morlet wavelet [2] (or at least its "standard" implementation in `morlet()`), which I guess is where the confusion comes from.

The parameters `w` and `s` correspond to the radian frequency ("omega") of the modulation complex exponential, and the scaling parameter of the wavelet ([1], Eq. (4.31)), respectively. Since `morlet2()` was specifically implemented to work with `cwt()`, which requires the second argument of the wavelet function to be the "width" of the wavelet, I've called it "width" in `morlet2()`.

[_[1] S. Mallat. A Wavelet Tour of Signal Processing. Elsevier Science, 1999._](https://www.di.ens.fr/~mallat/papiers/WaveletTourChap1-2-3.pdf)

[_[2] Wikipedia: Morlet wavelet_](https://en.wikipedia.org/wiki/Morlet_wavelet)